src/ext/ddtrace.c

```diff
@@ -113,18 +113,29 @@ static void register_span_data_ce(TSRMLS_D) {
     zend_declare_property_null(ddtrace_ce_span_data, "metrics", sizeof("metrics") - 1, ZEND_ACC_PUBLIC TSRMLS_CC);
 }
 
+static void _disable_if_incompatible_sapi_detected(TSRMLS_D) {
+    if (strcmp("fpm-fcgi", sapi_module.name) == 0 || strcmp("apache2handler", sapi_module.name) == 0 ||
+        strcmp("cli", sapi_module.name) == 0 || strcmp("cli-server", sapi_module.name) == 0) {
+        return;
+    }
+    ddtrace_log_debugf("Incompatible SAPI detected '%s'; disabling ddtrace", sapi_module.name);
+    DDTRACE_G(disable) = 1;
+}
```

Comment:
As per our brief conversation in chat today, can you prefix this static function with `_dd_`?